lib/cadet_web/controllers/missions_controller.ex

```diff
@@ -0,0 +1,190 @@
+defmodule CadetWeb.MissionsController do
+  use CadetWeb, :controller
+
+  use PhoenixSwagger
+
+  swagger_path :index do
+    get("/missions")
+
+    summary("Get a list of all missions")
+
+    security([%{JWT: []}])
+
+    produces("application/json")
+
+    response(200, "OK", Schema.ref(:MissionsList))
+    response(401, "Unauthorised")
+  end
+
+  swagger_path :open do
+    get("/missions/open")
+
+    summary("Get a list of open missions")
+
+    security([%{JWT: []}])
+
+    produces("application/json")
+
+    response(200, "OK", Schema.ref(:MissionsList))
+    response(401, "Unauthorised")
+  end
+
+  swagger_path :show do
+    get("/missions/{missionId}")
+
+    summary("Get information about one particular mission.")
+
+    security([%{JWT: []}])
+
+    produces("application/json")
+
+    parameters do
+      missionId(:path, :integer, "mission id", required: true)
+    end
+
+    response(200, "OK", Schema.ref(:Mission))
+    response(400, "Missing parameter(s) or invalid missionId")
+    response(401, "Unauthorised")
+  end
+
+  swagger_path :questions do
+    get("/missions/{missionId}/questions")
```

Comment:
I don't think we need an API call for each question. Rather, the questions for a particular assessment can be provided with the API call for that assessment as an array. This helps us bind the right actions to the onClicks for next, prev, and submit (and when to render these buttons) of the workspace.

i.e. the `mission` model can supply us with questions as an array property.

Reply:
It looks like one query to `/missions/{missionId}/questions` returns all the questions. However, you would then need to hit the `/answer/{type}/id` endpoint once per question to get the user's previous submission. Since we need to be able to refresh questions individually, I'm thinking maybe an alternative is to have `/answer/{type}?id={id1}?id={id2}` return a JSON of 
```
{
  id1: {
    type: "mcq",
    answer: 4
  },
  id2: {
    type: "programming",
    answer: "console.log('hello world!');"
  }, ...
}
```

Reply:
+1, though I'd prefer to not have to specify what type of question (MCQ or programming) in the query string.

Reply:
You shouldn't need to specify the question type in the query. I guess it's not even necessary to include the type in the response. You would then need to check against the locally stored collection of questions to determine question type (and thus how to interpret the value).

Alternatively, for simplicity, we can represent MCQ answers as a string (specifically a character). So something like:
```
{
  id1: "d", // or "4"
  id2: "console.log('hello world!');"
}
```
@indocomsoft Thoughts?

Reply:
@tuesmiddt Yeah, sorry the `/answer/*` endpoints were my very early plans -- we should be able to unify the 2 types.

As for MCQ answer, I was thinking to maybe tag each MCQ choice associated with an MCQ question with a pseudo-id that will be used to identify them uniquely (combination of questionId + choice pseudo-id). Doesn't sound very nice to me -- let me know if any of you have any suggestion. @remo5000 @sreycodes @rrtheonlyone 

@ningyuansg would you then rather have the questions themselves embedded in the  `get("/missions/{missionId}")` endpoint? That way we can just merge the two.